client/src/App.js

```diff
@@ -1,29 +1,38 @@
-import React, { Component } from 'react';
-
-import SavedList from './Movies/SavedList';
-import MovieList from './Movies/MovieList';
-import Movie from './Movies/Movie';
+import React, { Component } from "react";
+import { Route } from "react-router-dom";
+import SavedList from "./Movies/SavedList";
+import MovieList from "./Movies/MovieList";
+import Movie from "./Movies/Movie";
 
 export default class App extends Component {
-  constructor() {
-    super();
-    this.state = {
-      savedList: []
-    };
-  }
+	constructor() {
+		super();
+		this.state = {
+			savedList: []
+		};
+	}
 
-  addToSavedList = movie => {
-    const savedList = this.state.savedList;
-    savedList.push(movie);
-    this.setState({ savedList });
-  };
+	addToSavedList = movie => {
+		const savedList = this.state.savedList;
+		if (!savedList.map(savedMovie => savedMovie.id).includes(movie.id)) {
+			savedList.push(movie);
+		}
+		this.setState({ savedList });
```

Comment:
I think you're actually better off with the setState call inside the if block. Right now setState is being called whether savedList got changed or not, but it only needs to be called if savedList is being changed.